### PR TITLE
{Compute} Apply UsageError to replace CLIError prefixed with 'Usage error'

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**  

This PR uses the UsageError to replace the all the CLIErrors prefixed by 'Usage error: ' in vm command group, so that
1. the errors can be classified into userfaults
2. we can control whether to print error types in CLI Core instead of in each of the error messages

**Testing Guide**  

No difference from users' experience.

### CODE REVIEW

**Comment by yonzhan** [comment]
Besides Compute, other services will also do the similar changes, right?

**Review by yungezz — APPROVED** [review]

**Comment by houk-ms** [comment]
> Besides Compute, other services will also do the similar changes, right?

Yes, exactly. This is the most simple case where we can take to replace the CLIErrors.

File: src/azure-cli/azure/cli/command_modules/vm/_image_builder.py [hunk 1]
```diff
@@ -326,7 +327,7 @@ def process_img_tmpl_output_add_namespace(cmd, namespace):
     if len(outputs) != 1:
         err = "Supplied outputs: {}".format(outputs)
         logger.debug(err)
-        raise CLIError("Usage error: must supply exactly one destination type to add. Supplied {}".format(len(outputs)))
+        raise UsageError("must supply exactly one destination type to add. Supplied {}".format(len(outputs)))
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/vm/_image_builder.py:330]:
```suggestion
        raise UsageError("Must supply exactly one destination type to add. Supplied {}".format(len(outputs)))
```

**Review by qwordy — CHANGES_REQUESTED** [review]
Capitalize the first letter in all error messages.

File: src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
 from knack.log import get_logger
 from knack.util import CLIError
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:15]: Can we remove this line since it is not used any more?

Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:15]: good catch!

**Comment by houk-ms** [comment]
> Capitalize the first letter in all error messages.
You're right, modified.

**Review by qwordy — APPROVED** [review]

**Comment by yungezz** [comment]
hi @houk-ms will this change go into this release?

**Comment by houk-ms** [comment]
> hi @houk-ms will this change go into this release?

I prefer not. 

Reason is I am not quite sure whether the changes are necessary. I mean for the existing `CLIError` in our codes, i don't have a clear plan about how to deal with them.

**Review by qwordy — APPROVED** [review]

**Pull request closed without merging**